installer/installer.py

```diff
@@ -522,6 +537,98 @@ def cmd_install_netmodules(public_slave=False):
     return
 
 
+
+def cmd_install_cni(public_slave,
+                    cni_plugins_dir,
+                    cni_conf_dir):
+    """
+    Install Calico's CNI plugin.  A successful completion of the task
+    indicates successful installation
+
+    :param public_slave: Flag indicating if this is a public slave.
+    """
+    # Load the current Calico install info for CNI
+    install_config = load_config(CNI_INSTALL_CONFIG)
+
+    # Before starting the install, check that we are able to locate the agent
+    # process - if not there is not much we can do here.
+    if not install_config:
+        _log.debug("Have not started installation yet")
+        # noinspection PyTypeChecker
+        agent_process = wait_for_process(AGENT_EXE_RE,
+                                         AGENT_EXE_PARMS_RE,
+                                         MAX_TIME_FOR_AGENT_RESTART,
+                                         fail_if_not_found=False)
+        if not agent_process:
+            _log.info("Cannot find agent process - do not update config")
+            return
+
+        mesos_version, _, _ = get_host_info()
+        if not mesos_version:
+            _log.error("Mesos version does not support CNI. Performing a no-op.")
+            return
+
+        install_config["agent-created"] = None
+        install_config["cni-bin-installed"] = None
+        install_config["cni-conf-installed"] = None
+        store_config(CNI_INSTALL_CONFIG, install_config)
+
+    if not install_config.get("cni-bin-installed"):
+        # Install CNI binary
+        move_file_if_missing("./calico", cni_plugins_dir)
+        install_config["cni-bin-installed"] = True
+        store_config(CNI_INSTALL_CONFIG, install_config)
+
+        # Install Calico-IPAM
```

Comment:
This should be _before_ for store the config indicating that this was done...   i.e. it should be alongside the Install CNI binary line.